src/App.js

```diff
@@ -80,32 +81,39 @@ const App = (props) => {
         } catch (e) {
             chrome.hideGlobalFilter();
         }
+
+        setHasSystemsReady(true);
     }
 
     useEffect(() => {
         initChrome();
         // eslint-disable-next-line react-hooks/exhaustive-deps
     }, []);
 
+    if (!arePermissionsReady || !hasSystemsReady) {
+        return <PageLoading />;
+    }
+
+    if (hasSystemsReady && !hasSystems) {
+        return <ZeroState />;
+    }
+
     return (
-        arePermissionsReady ?
-            <PermissionContext.Provider
-                value={ {
-                    customPolicies: permissions.customPolicies,
-                    compliance: permissions.compliance,
-                    drift: permissions.drift,
-                    advisor: permissions.advisor,
-                    remediations: permissions.remediations,
-                    patch: permissions.patch,
-                    vulnerability: permissions.vulnerability,
-                    subscriptions: permissions.subscriptions,
-                    ros: permissions.ros,
-                    notifications: permissions.notifications,
-                    hasSystems
-                } }>
-                <DashboardRoutes childProps={ props } />
-            </PermissionContext.Provider>
-            : <PageLoading />
+        <PermissionContext.Provider
```

Comment:
It would make sense to extract this into a `PermissionsProvider`, with the `permissions` state (and what sets it).